### ISSUE TITLE
Display current simulation status

### ISSUE THREAD

**Issue by sriharshakandala** [CliMA/ClimaAtmos.jl#1976]
Display current simulation status at a user prescribed frequency from the root process. 
To start with, this could contain basic information like # of time steps simulated, total # of time steps, elapsed wall clock time and a timestamp at the time of print out. 
This will help the user understand how far the simulation has progressed.
This could be expanded to contain more useful information like `CFL max in the domain`, macro information like integrals of density, energy, momentum, etc.

cc: @szy21 , @akshaysridhar

**Comment by akshaysridhar** [comment]
Thanks @sriharshakandala: 

In previous iterations of ClimateMachine we had something like 
```
        """
        Update
            simtime = %s
            wallclock = %s
            efficiency (simtime / wallclock) = %s
            wallclock end (estimated) = %s
            norm(Q) = %.16e""",
```

a `SummaryLogCallback` : https://github.com/CliMA/ClimateMachine.jl/blob/2e0b6b7d97719e410d12a8596c98d5db7f891dbf/src/Driver/Callbacks/Callbacks.jl#L51C55-L51C55

and additional callbacks that `check_cons` (check conservation), `monitor_courant` (monitor courant numbers) 
```
================================================
                Courant numbers at simtime: %8.2f
                Δt = %8.2f s
                ------------------------------------------------
                Acoustic (vertical) Courant number    = %.2g
                Acoustic (horizontal) Courant number  = %.2g
                ------------------------------------------------
                Advection (vertical) Courant number   = %.2g
                Advection (horizontal) Courant number = %.2g
                ------------------------------------------------
                Diffusion (vertical) Courant number   = %.2g
                Diffusion (horizontal) Courant number = %.2g
                ================================================
```
and a more involved `StateCheck` Callback option as in https://github.com/CliMA/ClimateMachine.jl/blob/2e0b6b7d97719e410d12a8596c98d5db7f891dbf/src/Diagnostics/Debug/StateCheck.jl#L119 

I guess the intended display statement here incorporates some combination of these?

**Comment by Sbozzolo** [comment]
This was implemented in #2428
